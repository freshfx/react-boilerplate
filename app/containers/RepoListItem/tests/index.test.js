import React from 'react'
import {render} from '@testing-library/react'
import {IntlProvider} from 'react-intl'

import {RepoListItem} from '../index'

jest.mock('components/ListItem', () => ({item}) => item)

const id = 'repository-id-1'
const renderComponent = (props = {}) => render(
  <IntlProvider locale="en">
    <RepoListItem id={id} {...props} />
  </IntlProvider>
)

const defaultProps = {
  fullName: 'react-boilerplate/react-boilerplate',
  name: 'react-boilerplate',
  openIssuesCount: 20,
  ownerUsername: 'mxstbr',
  url: 'https://github.com/react-boilerplate/react-boilerplate'
}

describe('RepoListItem', () => {
  it('should match snapshot', () => {
    const props = {
      ...defaultProps,
      username: defaultProps.ownerUsername
    }
    const {container} = renderComponent(props)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should add nameprefix if user is not currentUser', () => {
    const {getByText} = renderComponent(defaultProps)
    expect(getByText(`${defaultProps.ownerUsername}/${defaultProps.name}`))
  })
})
