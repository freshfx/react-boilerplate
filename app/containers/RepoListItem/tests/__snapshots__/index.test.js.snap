// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoListItem should match snapshot 1`] = `
.c3 {
  fill: #ccc;
  margin-right: 0.25em;
}

.c2 {
  color: #41addd;
  height: 100%;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2:hover {
  color: #6cc0e5;
}

.c1 {
  color: #41addd;
  height: 100%;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c1:hover {
  color: #6cc0e5;
}

.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
}

<div
  class="c0"
>
  <a
    class="c1"
    href="https://github.com/react-boilerplate/react-boilerplate"
    target="_blank"
  >
    react-boilerplate
  </a>
  <a
    class="c2"
    href="https://github.com/react-boilerplate/react-boilerplate/issues"
    target="_blank"
  >
    <svg
      class="c3"
      height="1em"
      width="0.875em"
    >
      <path
        d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7S10.14 13.7 7 13.7 1.3 11.14 1.3 8s2.56-5.7 5.7-5.7m0-1.3C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7S10.86 1 7 1z m1 3H6v5h2V4z m0 6H6v2h2V10z"
      />
    </svg>
    20
  </a>
</div>
`;
